sql/plan/alter_foreign_key.go

```diff
@@ -244,7 +244,22 @@ func ResolveForeignKey(ctx *sql.Context, tbl sql.ForeignKeyTable, refTbl sql.For
 		col := cols[strings.ToLower(fkDef.Columns[i])]
 		parentCol := parentCols[strings.ToLower(fkDef.ParentColumns[i])]
 		if !col.Type.Equals(parentCol.Type) {
-			return sql.ErrForeignKeyColumnTypeMismatch.New(fkDef.Columns[i], fkDef.ParentColumns[i])
+			// There seems to be a special case where CHAR/VARCHAR/BINARY/VARBINARY can have unequal lengths.
```

Comment:
Should extract a validateCompatibleForeignKeyParentType method for these two identical blocks